sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/ExecutableLinesWalker.cs

```diff
@@ -96,43 +95,71 @@ private bool FindExecutableLines(SyntaxNode node)
 
                 case SyntaxKind.StructDeclaration:
                 case SyntaxKind.ClassDeclaration:
-                    var typeDeclarationSyntax = (BaseTypeDeclarationSyntax)node;
-                    return !HasExcludedCodeAttribute(typeDeclarationSyntax.AttributeLists);
+                    return !HasExcludedCodeAttribute((BaseTypeDeclarationSyntax)node, btdc => btdc.AttributeLists,
+                        canBePartial: true);
 
                 case SyntaxKind.MethodDeclaration:
-                    var methodSyntax = (BaseMethodDeclarationSyntax)node;
-                    return !HasExcludedCodeAttribute(methodSyntax.AttributeLists);
+                case SyntaxKind.ConstructorDeclaration:
+                    return !HasExcludedCodeAttribute((BaseMethodDeclarationSyntax)node, bmds => bmds.AttributeLists,
+                        canBePartial: true);
 
                 case SyntaxKind.PropertyDeclaration:
-                    var propertySyntax = (PropertyDeclarationSyntax)node;
-                    return !HasExcludedCodeAttribute(propertySyntax.AttributeLists);
-
                 case SyntaxKind.EventDeclaration:
-                    var eventSyntax = (EventDeclarationSyntax)node;
-                    return !HasExcludedCodeAttribute(eventSyntax.AttributeLists);
+                    return !HasExcludedCodeAttribute((BasePropertyDeclarationSyntax)node, bpds => bpds.AttributeLists);
+
+                case SyntaxKind.AddAccessorDeclaration:
+                case SyntaxKind.RemoveAccessorDeclaration:
+                case SyntaxKind.SetAccessorDeclaration:
+                case SyntaxKind.GetAccessorDeclaration:
+                    return !HasExcludedCodeAttribute((AccessorDeclarationSyntax)node, ads => ads.AttributeLists);
 
                 default:
                     return true;
             }
         }
 
-        private bool HasExcludedCodeAttribute(SyntaxList<AttributeListSyntax> attributeSyntaxLists)
+        private bool HasExcludedCodeAttribute<T>(T node, Func<T, SyntaxList<AttributeListSyntax>> getAttributeLists,

```

Comment:
Why not type the second parameter as _SyntaxList{AttributeListSyntax}_ directly, rather than as function?

Reply:
That's debatable :)  It does adds unnecessary complexity to this method declaration, and it'll be slower at runtime, although only by a small amount.